lib/speech/operation.js

```diff
@@ -0,0 +1,187 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*!
+ * @module speech/operation
+ */
+
+'use strict';
+
+var events = require('events');
+var modelo = require('modelo');
+
+var OPERATION_FIELDS = [
+  'name',
+  'metadata',
+  'done',
+  'error',
+  'response'
+];
+
+function pick(dest, src) {
+  OPERATION_FIELDS.forEach(function(field) {
+    dest[field] = src[field];
+  });
+}
+
+/**
+ * @type {module:common/grpc-service-object}
+ * @private
+ */
+var GrpcServiceObject = require('../common/grpc-service-object.js');
+
+/*! Developer Documentation
+ *
+ * @param {module:speech} speech - Speech object.
+ * @param {string} name - Name of the operation.
+ */
+/**
+ * An Operation object allows you to interact with a Google Cloud Speech
+ * operation.
+ *
+ * @constructor
+ * @alias module:speech/operation
+ *
+ * @example
+ * var speech = gcloud.speech({
+ *   projectId: 'grape-spaceship-123'
+ * });
+ *
+ * var operation = speech.operation('6885083136681374325');
+ */
+function Operation(service, values) {
+  this.service = service;
+  pick(this, values);
+
+  GrpcServiceObject.call(this, {
+    parent: service,
+    id: this.name,
+    methods: {}
+  });
+
+  events.EventEmitter.call(this);
+
+  this.completeListeners = 0;
+  this.hasActiveListeners = false;
+
+  this.listenForEvents_();
+}
+
+modelo.inherits(Operation, GrpcServiceObject, events.EventEmitter);
+
+/**
+ * Gets the latest state of a long-running operation. Clients can use this
+ * method to poll the operation result at intervals as recommended by the
+ * API service.
+ *
+ * @resource [Operations: GetOperation API Documentation]{@link https://cloud.google.com/speech/reference/rpc/google.longrunning#operations}
+ * @resource [Operations: Operation API Documentation]{@link https://cloud.google.com/speech/reference/rpc/google.longrunning#operation}
+ *
+ * @param {function} callback - The callback function.
+ * @param {?error} callback.err - An error returned while making this request.
+ * @param {object} callback.operation - The Operation object with its fields
+ *     populated.
+ * @param {object} callback.apiResponse - Raw API response.
+ *
+ * @example
+ * var operation = speech.operation('6885083136681374325');
+ * operation.get(function(err, operation, apiResponse) {
+ *   console.log(operation);
+ * });
+ */
+Operation.prototype.get = function(callback) {
+  var self = this;
+  var protoOpts = {
+    service: 'Operations',
+    method: 'getOperation'
+  };
+
+  this.service.request(protoOpts, {
```

Comment:
Just a style thing, but can you move this second arg (`{ name: this.name }`) to an object labeled `reqOpts`. It makes it easier to differentiate between the argument objects for a future contributor.